packages/server/src/db/index.js

```diff
@@ -1040,12 +1040,112 @@ async function inTenant(tenantId, agencyIds) {
     return result.same_tenant === true;
 }
 
+/**
+ * Creates and saves a new saved search, given a name, agency ID, user ID, and criteria
+ *  interface SavedSearch {
+    id?: number
+    name: string
+    agencyId: number
+    userId: number
+    criteria: string
+    createdAt?: string
+   }
+ * @param  SavedSearch           searchItem
+ * @return Promise<SavedSearch>
+ * */
+async function createSavedSearch(searchItem) {
+    const response = await knex
+        .insert({
+            name: searchItem.name,
+            agency_id: searchItem.agencyId,
+            created_by: searchItem.userId,
+            criteria: searchItem.criteria,
+        })
+        .into('grants_saved_searches')
+        .returning('*');
+
+    return {
+        id: response[0].id,
+        name: response[0].name,
+        agencyId: response[0].agency_id,
+        createdBy: response[0].created_by,
+        criteria: response[0].criteria,
+        createdAt: new Date(response[0].created_at).toISOString(),
+    };
+}
+
+/**
+ * Retrieves saved searches, given an agency ID, result limit, and result offset
```

Comment:
Aren't saved searches scoped by agency and user?

For example- User A has two saved searches. One with agency X and another with agency Y.
Usually the user is only viewing the dashboard for one agency at a time so most access patterns will be scoped by agency anyway.

Reply:
I took another look back at the #674 to refresh myself a bit on the expectations for this deliverable. The implementation in this PR is consistent with that ticket, but I think it might be worth checking in with @caitlinwinner / UX given the amount of time that elapsed between with that ticket was initially planned/when design was finalized/now. A lot of the recent conversations have explicitly touched on the idea that searches are 100% personal for this milestone and not shared in any way between users, in part due to the amount of design/UX considerations around having multiple people being able to edit the same batch of saved searches, as well as the fact that saved searches will be used to drive email notifications.

Based on the concrete implementation in this PR, I could see three options up for consideration, along with the following implications for each option:
1. **Saved searches are scooped only to users (no agency relationship whatsoever)**
    - Regardless of the agency the user has selected (is "active") in the UI, the user will always see the same set of saved searches.
2. **Saved searches are scoped primarily to the agency, and the user relationship is just for tracking authorship**
    - **This is the behavior currently implemented by this PR.**
    - All users who are a part of an agency will share the same set of saved searches. 
    - Any user deleting a saved search will delete the saved search for everyone else in the agency.
    - All users with grant digest notifications enabled will receive notifications for all saved searches that exist for their agency.
3. **Saved searches are scoped primarily to the user, and the agency is just for controlling visibility**
    - Saved searches are not shared / visible across multiple users within the same agency.
    - Deleting a saved search will only delete the saved search for the user who created it.
    - Users with grant digest notifications enabled will receive notifications only for the saved searches that they have created.
    - Switching between agencies changes the set of available saved searches. If the user makes use of the same search criteria across agencies, they need to create the search multiple times.
        - Caveat: unless we are "bundling" grants into a single daily digest email (which gives us the opportunity to deduplicate grants), users with identical search criteria will trigger duplicative grant notifications. It's been noted that this is also true for users with more than one saved search that match the same grant, but an additional argument in favor of ensuring that users are notified about grants in the form of a digest rather than receiving one-off emails.

Reply:
imo we can rule out 1
- The way I understand search criteria is that it is inherently agency-specific. Example: if I create keywords "fire emergency" for the fire-department agency, it seems unlikely I'd want to use this same criteria with another agency.

For 2 and 3, I'm not seeing how the backend crud implementation changes. We'll still need to retrieve data by user/agency, but we may end up putting some restrictions on top of these functions on who can delete/view the data returned.

Reply:
If you think there's future-proofing value in maintaining a direct relationship between a saved search and an agency (as opposed to an implicit relationship that's determined by traversing the user relationship, e.g. `saved_search.user.agency`) then I think it's okay to leave it in at this point. My hesitation derives from not having a clear picture at this point of what future feature(s) we're facilitating by having a direct relationship exist and therefore preferring a more minimal implementation, but I don't think there's much risk of us shooting ourselves in the foot by keeping it either.

Reply:
Quick nit suggestion- the parameter docstrings and the function signature already reflect that this function is user-scoped; I think it would be helpful to include that in the one-liner summary as well.

GitHub isn't letting me submit this as a suggestion for some reason, but I'm suggesting this tweak to line 1078 (i.e. add `for a user`):
```
 * Retrieves saved searches for a user, given an agency ID, result limit, and result offset
```